### PR TITLE
Implement RPC getErrorHint

### LINKED ISSUE #1253: Implement RPC to get an error hint
Currently, the failed invoice contains RuntimeError to indicate the reason for failure.
However, this makes maintaining code and revising the spec difficult because the order of verification should be kept.
Thus, I propose to remove RuntimeError from an invoice. This will simply make the invoice a boolean.

And it should add a new RPC that returns the RuntimeError that the node has. I think the specification below will suffice.

---
## chain_getErrorHint
Gets a hint to find out why the transaction failed.
It returns `null` if the transaction is not involved in the chain or succeeded.
If the transaction failed, this should return the reason for the transaction failing.
It's better to return the human-readable string; however, the message is not guaranteed.

Other nodes may return a different message at the same time.
And it does not guarantee the consistency of the values returned by the same node.

### Params
 1. transaction hash - `H256`
 2. block number: `number` | `null`

### Returns
`null` | `string` - `null` if there is no hint, `string` if the transaction failed.

Errors: `Invalid Params`

### PR DESCRIPTION
**Pull request by HoOngEe**
Implement new RPC which get error string from invoice.

It closes #1253

### CODE REVIEW

File: core/src/client/mod.rs [hunk 1]
```diff
@@ -245,7 +245,6 @@ pub trait BlockChainClient:
     /// Get block status by block header hash.
     fn block_status(&self, id: &BlockId) -> BlockStatus;
 
-
```
Comment (sgkim126) [inline core/src/client/mod.rs]: Unrelated change

Comment (HoOngEe) [reply core/src/client/mod.rs]: Now that change is removed

File: rpc/src/v1/impls/chain.rs [hunk 2]
```diff
@@ -189,6 +189,17 @@ where
         Ok(self.client.balance(address, block_id.into()).map(Into::into))
     }
 
+    fn get_error_hint(&self, transaction_hash: H256) -> Result<Option<String>> {
+        let invoice_op = self.client.parcel_invoice(&transaction_hash.into());
```
Comment (sgkim126) [inline rpc/src/v1/impls/chain.rs]: This can be simplified as below.
```rust
        if let Some(Invoice::Failure(error_string)) = self.client.parcel_invoice(&transaction_hash.into()) {             
            Ok(Some(error_string))                                                                                       
        } else {                                                                                                         
            Ok(None)                                                                                                     
        }                                                                                                                
```

Comment (HoOngEe) [reply rpc/src/v1/impls/chain.rs]: Simplified.
